src/HERMIT/API/Types.hs

```diff
@@ -89,15 +93,27 @@ proxyToJSON Proxy = String $ pack $ show $ typeOf (undefined :: a)
 class FromJSON a => Response a where
   printResponse :: a -> IO ()
 
+quietModeM :: IORef Bool
+quietModeM = unsafePerformIO (newIORef False)
+{-# NOINLINE quietModeM #-}
+
 instance Response () where
   printResponse () = return ()
 
 instance Response String where
-  printResponse = print
+  printResponse str = do
+    quietMode <- readIORef quietModeM
+    if quietMode
+      then return ()
+      else print str
```

Comment:
Consider refactoring lines like this into `when quietMode $ print str`. (Also, convert occurrences of `when debug` to `when quietMode`.)